### ISSUE TITLE
Maven Functions Plugin - didn't pick up runtime scoped dependencies

### ISSUE THREAD

**Issue by saturnism** [GoogleCloudPlatform/functions-framework-java#70]
```
INFO] com.example:container-analysis-function:jar:0.0.1-SNAPSHOT
[INFO] +- com.google.cloud.functions:functions-framework-api:jar:1.0.1:provided
[INFO] +- com.google.cloud:google-cloud-containeranalysis:jar:1.1.0:compile
[INFO] |  +- io.grpc:grpc-api:jar:1.30.2:compile
...
[INFO] |  +- io.perfmark:perfmark-api:jar:0.19.0:runtime
...
[INFO] |  \- com.google.api.grpc:proto-google-iam-v1:jar:0.13.0:compile
[INFO] +- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] \- org.projectlombok:lombok:jar:1.18.12:compile
```

Notice that `perfmark-api` is in the `runtime` scope.
When running the function w `./mvnw function:run`, and trigger the container analysis code path, it throws:
```
Caused by: java.lang.NoClassDefFoundError: io/perfmark/PerfMark
	at io.grpc.internal.ClientCallImpl.<init>(ClientCallImpl.java:109)
	at io.grpc.internal.ManagedChannelImpl$RealChannel.newCall(ManagedChannelImpl.java:890)
	at io.grpc.internal.ServiceConfigInterceptor.interceptCall(ServiceConfigInterceptor.java:137)
	at io.grpc.ClientInterceptors$InterceptorChannel.newCall(ClientInterceptors.java:156)
```

However, the function works when deployed to the server.

Adding the `perfmark-api` as a direct dependency in `compile` scope made the maven plugin work.

**Comment by eamonnmcmanus** [comment]
Just checking, are you using the latest version of the plugin? This looks like something that was fixed by #32.

**Comment by saturnism** [comment]
doh, it was 0.9.3 confirmed 0.9.5 worked.
